workflow-runtime/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt

```diff
@@ -239,6 +261,10 @@ internal class WorkflowNode<PropsT, StateT, OutputT : Any, RenderingT>(
     // Tear down workflows and workers that are obsolete.
     subtreeManager.commitRenderedChildren()
     workers.commitStaging { it.channel.cancel() }
+    // Side effect jobs are launched lazily, since they can send actions to the sink, and must not
+    // be started until after context is frozen.
+    sideEffects.forEachStaging { it.job.start() }
```

Comment:
Add tests for multiple side effects (both not started until after render) and to ensure side effects with different keys are handled independently.
- both started in same render pass
- staggered
  - overlapping
  - non-overlapping